script.js

```diff
@@ -1,4 +1,341 @@
+    gameMode = dealerTurn;
+    return `dealer's turn, dealer has ${dealerHand.length} cards <br><br> please click submit to continue the game `;
+  }
+  // check initial hand of 2 cards for black jack condition
+
+  if (
+    ((playerHand[0].rank == 1 || playerHand[1].rank == 1) &&
+      (playerHand[0].rank >= 10 || playerHand[1].rank >= 10)) ||
+    playersTotalPoints == 21
+  ) {
+    return `BlackJack for player <br><br> player hand ${printPlayerHand()} <br><br> player points ${checkPlayerPoint()} <br><br> enter 'stand' to continue `;
+  }
+  // busted if point is above 21
+  else if (playersTotalPoints > 21) {
+    return `busted <br><br> player hand  ${printPlayerHand()} <br><br> player points ${checkPlayerPoint()} <br><br> please enter 'stand' to continue the game`;
+  }
+  // input validation for hit and stand
+  else if (playerChoice !== hit && playerChoice !== stand) {
+    return `enter 'hit or 'stand' only <br><br> player Hand ${printPlayerHand()} <br><br>player points ${checkPlayerPoint()}`;
+  }
+  // if player choose hit
+  else if (playerChoice == hit && playersTotalPoints <= 21) {
+    playerHand.push(shuffledDeck.pop());
+    return `player hand ${printPlayerHand()}<br><br> player points ${checkPlayerPoint()}<br><br> please choose hit or stand`;
+  }
+};
+
+// function if dealer choose to hit or stand
+
+var dealerHitOrStand = function () {
+  checkDealerPoint();
+  // if dealer has black jack switch game mode
+  if (
+    ((dealerHand[0].rank == 1 || dealerHand[1].rank == 1) &&
+      (dealerHand[0].rank >= 10 || dealerHand[1].rank >= 10)) ||
+    dealersTotalPoints == 21
+  ) {
+    gameMode = checkWinner;
+    return `black jack for dealer <br><br> dealer hand  ${printDealerHand()}<br><br> dealer points ${checkDealerPoint()}`;
+  }
+
+  // busted if points > 21
+  if (dealersTotalPoints > 21) {
+    gameMode = checkWinner;
+    return `dealer busted<br><br> dealer hand  ${printDealerHand()}<br><br> dealer points ${checkDealerPoint()}`;
+  }
+  // dealer draws a card if points are less than 17
+  if (dealersTotalPoints <= 15) {
+    dealerHand.push(shuffledDeck.pop());
+    return `dealer draws a card dealer has${dealerHand.length} cards <br><br> click submit to continue the game`;
+  }
+  // dealer stand if points are greater than 17 and less or equal to 21
+  else if (dealersTotalPoints <= 21) {
+    gameMode = checkWinner;
+    return `dealer stand <br><br> dealer hand  ${printDealerHand()} <br><br> dealer points ${checkDealerPoint()} <br><br> click submit to continue the game`;
+  }
+};
+
+// main function
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValu = ``;
+  if (gameMode == starting) {
+    // switch gmae mode
+    gameMode = dealCard;
+    generateDeck();
+    return `welcome ${player} please click "submit" to start the game`;
+  }
+  if (gameMode == dealCard) {
+    // deal 2 cards for each of the players
+    drawCard(shuffleCards(cardDeck));
```

Comment:
if you do it this way, you will be shuffling the deck whenever you draw a card. In the real game, you should only shuffle the deck once before the start of the game, then deal from the shuffled deck subsequently